frontend/lint/tests/hacspec.rs

```diff
@@ -0,0 +1,54 @@
+//! Tests for the hacspec linter
+
+use std::process::Command;
+
+mod util;
+use util::*;
+
+const TESTS: [Test; 2] = [
+    Test {
+        stderr: "",
+        manifest_path: "lint/hacspec_tests/v1-lib/Cargo.toml",
+    },
+    Test {
+        stderr: "warning: [Circus] Mutable arguments are not supported
+ --> mut_arg/src/lib.rs:1:23
+  |
+1 | pub fn add(left: &mut usize, right: usize) {
+  |                       ^^^^^
+
+warning: `mut_arg` (lib) generated 1 warning",
+        manifest_path: "lint/hacspec_tests/mut_arg/Cargo.toml",
+    },
+];
+
+#[test]
+fn lint_hacspec() {
+    install_driver();
+
+    for test in TESTS {
+        let mut cmd = Command::new("cargo");
+        cmd.current_dir("../");
+
+        let cmd = cmd.args(&[
+            "run",
+            "--bin",
+            "cargo-circus",
+            "--",
+            "-C",
+            "--manifest-path",
+            test.manifest_path,
+            ";",
+            "lint",
+            "hacspec",
+        ]);
```

Comment:
`cargo run` calls `cargo-circus` with env vars `DYLD_FALLBACK_LIBRARY_PATH` and `LD_LIBRARY_PATH` set to specific values, which messes up things in a rustup-free environment

Reply:
So `cargo run` doesn't work without rustup here? But I think it's generally fine to require `rustup` for this repo.

Reply:
Yes, without rustup something is going wrong.
Well, I fixed it in 42c2744c8fb806ebc10baaeeb2dc8d75db5947c1 by adding an environment variable the circus driver is sensible to, so that we can inject the absolute path to the local cargo binary

If that change is fine to you, feel free to cherry-pick it in this branch :) (I can also do it if you want)

Reply:
Yes, without rustup something goes wrong.

42c2744c8fb806ebc10baaeeb2dc8d75db5947c1 fixes this issue as well, by explicitly remove such env vars Cargo adds